arch/x86/adler32_avx512_p.h

```diff
@@ -3,6 +3,17 @@
 
 #include <immintrin.h>
 #include <stdint.h>
+
+/* Written because Visual C++ toolchains before v142 have constant overflow in AVX512 intrinsic macros */
+#if defined(_MSC_VER) && !defined(_MM_K0_REG8)
+#  undef _mm512_extracti64x4_epi64
+#  define _mm512_extracti64x4_epi64(v1, e1) _mm512_maskz_extracti64x4_epi64(UINT8_MAX, v1, e1)
+#  undef _mm512_set1_epi16
+#  define _mm512_set1_epi16(e1) _mm512_maskz_set1_epi16(UINT32_MAX, e1)
+#  undef _mm512_maddubs_epi16
+#  define _mm512_maddubs_epi16(v1, v2) _mm512_maskz_maddubs_epi16(UINT32_MAX, v1, v2)
```

Comment:
So is this actually a bad definition in their intrinsic that causes a partial mask, or is this just something silly like Microsoft defines this as a macro with -1 was the mask (which is technically correct but an overly pedantic compiler might try to complain about it)?

Reply:
@KungFuJesus Microsoft forgot to define the mask for half of the variants, and instead used mask for twice the big bit width and assumed their own compiler will truncate it without warning. So basically there was only `_MM_K0_REG16` and `_MM_K0_REG64` defined.

Reply:
Ah, I see, so it sounds like they weren't quite anticipating AVX512BW and just dropped in the 16 bit masks instead.  Which still result in correct behavior, but the implicit truncation causes their pedantic warnings to trigger.  It is still silly but I guess functionally this is fine.

Reply:
@KungFuJesus Microsoft likes to do things their own way and not copy how others have done the same thing... I don't think they anticipated that developers start to use higher warning levels than 3 on production code or that developers want to treat all warnings as errors, even though they anticipated that developers want to silence some warnings or to move some warnings to different level.